lib/live_view_native/tag_engine.ex

```diff
@@ -17,5 +18,30 @@ defmodule LiveViewNative.TagEngine do
 
   def classify_type(name), do: {:tag, name}
 
+  @impl true
+  def preprocess_tokens(tokens) do
+    Enum.map(tokens, &expand_template_attributes/1)
+  end
+
+  @impl true
   def void?(_), do: false
+
+  ###
+  defp expand_template_attributes({:tag, element, [_ | _] = attrs, metadata}) do
+    {:tag, element, Enum.map(attrs, &expand_template_attribute/1), metadata}
+  end
+
+  defp expand_template_attributes(token), do: token
+
+  defp expand_template_attribute({attr_name, nil, metadata} = attr) do
+    if String.starts_with?(attr_name, "#") do
+      template_name = String.replace_leading(attr_name, "#", ":")
+
+      {"template", {:expr, template_name, metadata}, metadata}
+    else
+      attr
+    end
+  end
```

Comment:
This is a bit simpler and I think has the same behavior:
```suggestion
  defp expand_template_attribute({"#" <> rest, nil, metadata} = attr) do
    {"template", {:expr, ":" <> rest, metadata}, metadata}
  end
```